tests/manage/pv_services/pvc_snapshot/test_snapshot_at_different_pvc_utlilization_level.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import pytest
+from copy import deepcopy
+
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework.testlib import skipif_ocs_version, ManageTest, tier1
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_ocs_version('<4.6')
+@pytest.mark.polarion_id('OCS-2318')
+class TestSnapshotAtDifferentPvcUsageLevel(ManageTest):
+    """
+    Tests to take snapshot when PVC usage is at different levels
+    """
+    @pytest.fixture(autouse=True)
+    def setup(self, create_pvcs_and_pods):
+        """
+        Create PVCs and pods
+
+        """
+        self.pvc_size = 10
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=self.pvc_size,
+            access_modes_rbd=[constants.ACCESS_MODE_RWO],
+            access_modes_cephfs=[constants.ACCESS_MODE_RWO]
+        )
+
+    def test_snapshot_at_different_usage_level(
+        self, pod_factory, teardown_factory, snapshot_restore_factory
+    ):
+        """
+        Test to take multiple snapshots of same PVC when the PVC usage is at
+        0%, 20%, 40%, 60%, and 80%, then delete the parent PVC and restore the
+        snapshots to create new PVCs
+
+        """
+        snapshots = []
+        for usage in [0, 20, 40, 60, 80]:
+            if usage != 0:
+                for pod_obj in self.pods:
+                    log.info(
```

Comment:
not sure I understand where the `usage` takes effect in each iteration

Reply:
When usage is 0, IO should be skipped and proceed directly to snapshot. Usage value will change in each iteration of the outer loop.

Reply:
the only place `usage` var is being used at is the filename

Reply:
It is used to give appropriate logs and to skip fio in first iteration.

Reply:
discussed offline and now it's clear to me - in each iteration, 2GB is being added on top of the data that was written in the previous iterations